pkg/collect/safe_map.go

```diff
@@ -25,6 +25,19 @@ func (m *SafeMap) Get(k string) *Value {
 	return &Value{v, ok}
 }
 
+// Values returns all key-values stored in map
+func (m *SafeMap) Values() map[string]interface{} {
```

Comment:
If m.inner equals to nil, do we need to return nil instead of `map[string]interface{}`?

Just a little confused on this.

Reply:
m.inner can not be nil,  since
```
func NewSafeMap() *SafeMap {
 	return &SafeMap{
 		inner: make(map[string]interface{}, 16),
 	}
 }
```

Reply:
Someone has some potential possibility that they do not use NewSafeMap() to create a SafeMap. If they create it via `safetyMap := &SafeMap{}`, it works finely. But, if they call function `safetyMap.Values()`, it could panic. Right?

Reply:
The same reason with #1132 .
for range a nil map will never cause panic, since nil value will convert to a nil map(map[] ), that add a nil check is useless.